lib/wallaroo/core/router_registry/router_registry.pony

```diff
@@ -197,15 +205,28 @@ actor RouterRegistry
   be application_ready_to_work() =>
     _application_ready_to_work = true
 
+  be start_sources() =>
+    _sources_started = true
+    if not _recovery_protocol_complete then
+      for sl in _source_listeners.values() do
+        sl.start_sources()
+      end
+    end
+
   be recovery_protocol_complete() =>
     """
     Called when Recovery is finished. If we're not recovering, that's right
     away.
     """
-    _recovery_protocol_complete = true
-    for sl in _source_listeners.values() do
-      sl.recovery_protocol_complete()
+    // TODO [source-migration]: Ask John if we still want to start sources here
+    // given that source listeners are now initializable
```

Comment:
Let's talk about this.  I think we should no longer manage things outside in this way.  The SourceListener now knows exactly where it is in the initialization phases and can start things accordingly.

Reply:
This should allow us to remove the fiddly logic here.